File: static/js/raketka.js
let coefElement = document.getElementById('current-coef');
let realTimeCoef = document.getElementById('real-time-coef');
let stopButton = document.getElementById('stop-rocket');
let startButton = document.querySelector('#raketka-form button[type="submit"]');
let ctx = document.getElementById('flightChart').getContext('2d');
let flightData = {
    labels: [],
    datasets: [{
        label: 'Траектория полета ракеты',
        borderColor: 'rgb(97,203,109)',
        data: [],
        fill: false,
        tension: 0.3
    }]
};

let chart = new Chart(ctx, {
    type: 'line',
    data: flightData,
    options: {
        scales: {
            x: {
                title: {display: true, text: 'Время'}
            },
            y: {
                beginAtZero: true,
                title: {display: true, text: 'Коэффициент'}
            }
        }
    }
});

function resetFlightChart() {
    flightData.labels = [];
    flightData.datasets[0].data = [];
    chart.update();
}

let rocketInterval;
let currentCoef = 1.00;

function animateRocketFlight(crashCoef, autoStop, callback) {
    let t = 0;
    rocketInterval = setInterval(() => {
        t += 1;
        currentCoef = 1 + Math.pow(t / 100, 2);

        coefElement.innerHTML = `Коэффициент: ${currentCoef.toFixed(2)}`;
        realTimeCoef.innerHTML = currentCoef.toFixed(2);

        flightData.labels.push(t);
        flightData.datasets[0].data.push(currentCoef);
        chart.update();

        if (currentCoef >= autoStop || currentCoef >= crashCoef) {
            clearInterval(rocketInterval);
            stopButton.style.display = 'none';
            startButton.style.display = 'flex';
            callback(currentCoef);
        }
    }, 150);
}

document.getElementById('raketka-form').addEventListener('submit', function (event) {
    event.preventDefault();

    const bet = document.getElementById('bet').value;
    const auto_stop = parseFloat(document.getElementById('auto_stop').value);

    resetFlightChart();

    document.getElementById('result').innerHTML = '';

    stopButton.style.display = 'flex';
    startButton.style.display = 'none';

    fetch('/raketka/play', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify({bet: bet, auto_stop: auto_stop})
    })
        .then(response => response.json())
        .then(data => {
            const crashCoef = data.coef;

            animateRocketFlight(crashCoef, auto_stop, (finalCoef) => {
                document.getElementById('result').innerHTML = `
                    <p>${data.message}</p>
                    <p>Баланс: ${data.balance}</p>
                `;
            });
        });
});

stopButton.addEventListener('click', function () {
    clearInterval(rocketInterval);
    stopButton.style.display = 'none';
    startButton.style.display = 'flex';
    document.getElementById('result').innerHTML = `
            <p>Вы остановили ракету на коэффициенте ${currentCoef.toFixed(2)}</p>
            <p>Ваш выигрыш: ${(currentCoef * document.getElementById('bet').value).toFixed(2)} токенов</p>
        `;
});

File: static/py/raketka.py
import random

class RocketGame:
    def __init__(self):
        self.balance = 100 
        self.max_crash_coef = 5.00  
        self.min_bet = 10 
        self.min_auto_stop = 1.25

    def place_bet(self, bet_amount, auto_stop):

        if bet_amount < self.min_bet:
            return "Ставка слишком мала. Минимальная ставка — 10 токенов."
        if auto_stop < self.min_auto_stop:
            return "Автостоп слишком мал. Минимальный автостоп — 1.25."

        if bet_amount > self.balance:
            return "Недостаточно токенов для ставки."


        crash_at = random.triangular(1.25, self.max_crash_coef, 1.5) 
        crash_at = round(crash_at, 2)

        current_coef = 1.00 


        while current_coef < self.max_crash_coef:
            current_coef += 0.01
            current_coef = round(current_coef, 2)


            if current_coef >= auto_stop:
                win_amount = bet_amount * auto_stop
                self.balance += win_amount - bet_amount 
                return {
                    'result': 'win',
                    'message': f"Автостоп сработал! Вы выиграли {win_amount} токенов.",
                    'balance': self.balance,
                    'coef': auto_stop
                }

 
            if current_coef >= crash_at:
                self.balance -= bet_amount
                return {
                    'result': 'lose',
                    'message': f"Ракета сломалась на коэффициенте {crash_at}. Вы потеряли {bet_amount} токенов.",
                    'balance': self.balance,
                    'coef': crash_at
                }

        return "Игра окончена."